specification/web/resource-manager/Microsoft.Web/stable/2016-06-01/logicAppsManagementClient.json

```diff
@@ -0,0 +1,2059 @@
+        "ConnectionGatewayReference": {
+            "description": "The gateway reference",
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "name": {
+                    "description": "The resource reference name",
+                    "type": "string"
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/definitions/ResourceReference"
+                }
+            ]
+        },
+        "CustomApiDefinition": {
+            "type": "object",
+            "description": "A custom API",
+            "additionalProperties": false,
+            "properties": {
+                "properties": {
+                    "$ref": "#/definitions/CustomApiPropertiesDefinition"
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/definitions/ResourceDefinition"
+                }
+            ]
+        },
+        "CustomApiPropertiesDefinition": {
+            "type": "object",
+            "description": "Custom API properties",
+            "additionalProperties": false,
+            "properties": {
+                "connectionParameters": {
+                    "$ref": "#/definitions/InsensitiveDictionaryOfJToken"
+                },
+                "runtimeUrls": {
+                    "type": "array",
+                    "description": "Runtime URLs",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "capabilities": {
+                    "type": "array",
+                    "description": "The custom API capabilities",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "swagger": {},
+                "brandColor": {
+                    "type": "string",
+                    "description": "Brand color"
+                },
+                "description": {
+                    "type": "string",
+                    "description": "The custom API desciption"
+                },
+                "displayName": {
+                    "type": "string",
+                    "description": "The display name"
+                },
+                "iconUri": {
+                    "type": "string",
+                    "description": "The icon URI"
+                },
+                "backendService": {
+                    "$ref": "#/definitions/ApiResourceBackendService"
+                },
+                "apiDefinitions": {
+                    "$ref": "#/definitions/ApiResourceDefinitions"
+                },
+                "apiType": {
+                    "$ref": "#/definitions/ApiType"
+                },
+                "wsdlDefinition": {
+                    "$ref": "#/definitions/WsdlDefinition"
+                }
+            }
+        },
+        "InsensitiveDictionaryOfJToken": {
```

Comment:
why do we need this? (rather than simply using "type": "object")

Reply:
It was autogenerated from the tool I was using and turned out to be a pretty common so I just ended up using it. I can remove it if you would prefer

Reply:
we can also check what other files are doing. seems weird though (also not sure how the auto-generated docs will look)